power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -634,19 +795,42 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
                                                       ->transformer.tap_range());
             }
         }
+
+        total_iterations = 0;
+    }
+
+    void bs_prep(std::vector<std::vector<RegulatedTransformer>> const& regulator_order) {
+        if (tap_search_ == SearchMethod::linear_search) {
+            return;
+        }
+        if (!binary_search_.empty()) {
+            binary_search_.clear();
+        } else {
+            binary_search_.reserve(regulator_order.size());
+        }
```

Comment:
without overhead
```suggestion
        binary_search_.clear();
        binary_search_.reserve(regulator_order.size());
```

Reply:
The reset was for a different purpose, which turned out to be redundant. The if logic will be removed, clear too.